### ISSUE TITLE
Gradle warning about maven plugin

### ISSUE THREAD

**Issue by ls-andrew-goodale** [auth0/react-native-auth0#400]
### Describe the problem

When gradle initializes our React Native project with this library, it outputs the following warning:
```
> Configure project :react-native-auth0
The maven plugin has been deprecated. This is scheduled to be removed in Gradle 7.0. Please use the maven-publish plugin instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.7/userguide/upgrading_version_5.html#legacy_publication_system_is_deprecated_and_replaced_with_the_publish_plugins
        at build_12vjxjx9ti7of850055e9c6uh.run(/..../node_modules/react-native-auth0/android/build.gradle:18)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)
```

### What was the expected behavior?

This library's gradle code should not use deprecated Gradle APIs.

Is the maven plugin even necessary for end-users of this library?

### Reproduction

On the command line in a React Native project, run a gradle command such as `./gradlew clean`.

### Environment

- react-native-auth0 2.8.3
- React Native 0.64.2
- Gradle 6.7

**Comment by lbalmaceda** [comment]
This is the offending line https://github.com/auth0/react-native-auth0/blob/master/android/build.gradle#L18. I'm afraid is being used at the bottom of the file, referencing a local file. But I haven't pull the repo to try removing it. It could potentially be fixed by upgrading the project following these guidelines https://reactnative.dev/docs/upgrading or with what they recommend here https://docs.gradle.org/6.7/userguide/upgrading_version_5.html#legacy_publication_system_is_deprecated_and_replaced_with_the_publish_plugins.

I believe that for now the warning can be safely ignored as long as the Gradle version used stays below 7.

**Comment by bneigher** [comment]
Do you guys anticipate supporting Gradle 7?  Looks like react-native is moving to it with 0.67 so I imagine this becoming a blocking issue soon

**Comment by poovamraj** [comment]
This warning has been fixed in [v2.11.0](https://github.com/auth0/react-native-auth0/releases/tag/v2.11.0). You can check this [PR](https://github.com/auth0/react-native-auth0/pull/429) for reference. Closing this issue for now. Please feel free to reopen if the issue isn't fixed.

@bneigher We plan to support Gradle 7. Currently, #446 is blocking it and we are working on fixing this issue. You can follow that ticket for status update
